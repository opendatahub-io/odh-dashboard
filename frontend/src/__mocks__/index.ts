export * from './mockDashboardConfig';
export * from './mockDataSciencePipelinesApplicationK8sResource';
export * from './mockK8sResourceList';
export * from './mockProjectK8sResource';
export * from './mockRouteK8sResource';
export * from './mockStatus';
export * from './mockRunKF';
export * from './mockRecurringRunKF';
export * from './mockExperimentKF';
export * from './mockPipelineVersionsProxy';
export * from './mockPipelinesProxy';
export * from './mockDscStatus';
export * from './mockNotebookK8sResource';
export * from './mockNotebookState';
export * from './mockPipelineKF';
export * from './mockSecretK8sResource';
export * from './mockGoogleRpcStatusKF';
export * from './mockK8sStatus';
export * from './mockComponents';
export * from './mockRegisteredModel';
export * from './mockModelVersion';
export * from './mockModelVersionList';
export * from './mockModelArtifactList';
export * from './mockModelRegistryService';
export * from './mockServingRuntimeK8sResource';
export * from './mockInferenceServiceK8sResource';
export * from './mockStorageClasses';
export * from './mockConfigMapSecrets';
