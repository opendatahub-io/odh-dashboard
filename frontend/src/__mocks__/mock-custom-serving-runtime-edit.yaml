apiVersion: serving.kserve.io/v1alpha1
kind: ServingRuntime
metadata:
  name: template-1
  annotations:
    openshift.io/display-name: Updated Caikit
  labels:
    opendatahub.io/dashboard: 'true'
spec:
  builtInAdapter:
    memBufferBytes: 134217728
    modelLoadingTimeoutMillis: 90000
    runtimeManagementPort: 8888
    serverType: ovms
  containers:
    - args:
        - --port=8001
        - --rest_port=8888
        - --config_path=/models/model_config_list.json
        - --file_system_poll_wait_seconds=0
        - --grpc_bind_address=127.0.0.1
        - --rest_bind_address=127.0.0.1
        - --target_device=NVIDIA
      image: quay.io/modh/openvino-model-server@sha256:c89f76386bc8b59f0748cf173868e5beef21ac7d2f78dada69089c4d37c44116
      name: ovms
      resources:
        limits:
          cpu: '0'
          memory: 0Gi
        requests:
          cpu: '0'
          memory: 0Gi
  grpcDataEndpoint: port:8001
  grpcEndpoint: port:8085
  multiModel: true
  protocolVersions:
    - grpc-v1
  supportedModelFormats:
    - autoSelect: true
      name: openvino_ir
      version: opset1
    - autoSelect: true
      name: onnx
      version: '1'
