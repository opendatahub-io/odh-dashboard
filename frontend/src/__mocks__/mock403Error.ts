import { K8sStatus } from '~/k8sTypes';

export const mock403Error = ({ message = '403 Forbidden' }: Partial<K8sStatus>): K8sStatus => ({
  kind: 'Status',
  apiVersion: 'v1',
  status: 'Failure',
  message,
  reason: 'Forbidden',
  code: 403,
});
