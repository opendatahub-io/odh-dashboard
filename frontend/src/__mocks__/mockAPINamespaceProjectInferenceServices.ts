// api/k8s/apis/serving.kserve.io/v1beta1/namespaces/{project}/inferenceservices

export const mockAPINamespaceProjectInferenceServices = {
    "apiVersion": "serving.kserve.io/v1beta1",
    "items": [
        {
            "apiVersion": "serving.kserve.io/v1beta1",
            "kind": "InferenceService",
            "metadata": {
                "annotations": {
                    "openshift.io/display-name": "Model Name",
                    "serving.kserve.io/deploymentMode": "ModelMesh"
                },
                "creationTimestamp": "2023-03-17T16:12:41Z",
                "generation": 1,
                "labels": {
                    "name": "model-name",
                    "opendatahub.io/dashboard": "true"
                },
                "name": "model-name",
                "namespace": "project",
                "resourceVersion": "1309350",
                "uid": "c275044f-85f7-44dc-9539-b9fdf6fcf334"
            },
            "spec": {
                "predictor": {
                    "model": {
                        "modelFormat": {
                            "name": "onnx",
                            "version": "1"
                        },
                        "runtime": "model-server-project",
                        "storage": {
                            "key": "aws-connection-test",
                            "path": ""
                        }
                    }
                }
            },
            "status": {
                "conditions": [
                    {
                        "lastTransitionTime": "2023-03-17T16:12:41Z",
                        "status": "False",
                        "type": "PredictorReady"
                    },
                    {
                        "lastTransitionTime": "2023-03-17T16:12:41Z",
                        "status": "False",
                        "type": "Ready"
                    }
                ],
                "modelStatus": {
                    "copies": {
                        "failedCopies": 0
                    },
                    "lastFailureInfo": {
                        "message": "Waiting for runtime Pod to become available",
                        "modelRevisionName": "model-size__isvc-59ce37c85b",
                        "reason": "RuntimeUnhealthy"
                    },
                    "states": {
                        "activeModelState": "Pending",
                        "targetModelState": ""
                    },
                    "transitionStatus": ""
                }
            }
        }
    ],
    "kind": "InferenceServiceList",
    "metadata": {
        "continue": "",
        "resourceVersion": "1466368"
    }
}