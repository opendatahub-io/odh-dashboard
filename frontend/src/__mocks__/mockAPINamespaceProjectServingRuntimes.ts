// api/k8s/apis/serving.kserve.io/v1alpha1/namespaces/project/servingruntimes

export const mockAPINamespaceProjectServingRruntimes = {
    "apiVersion": "serving.kserve.io/v1alpha1",
    "items": [
        {
            "apiVersion": "serving.kserve.io/v1alpha1",
            "kind": "ServingRuntime",
            "metadata": {
                "creationTimestamp": "2023-03-17T16:05:55Z",
                "generation": 1,
                "labels": {
                    "name": "model-server-project",
                    "opendatahub.io/dashboard": "true"
                },
                "name": "model-server-project",
                "namespace": "project",
                "resourceVersion": "1302239",
                "uid": "170e3426-6596-4c0b-b21a-f36b039e7949"
            },
            "spec": {
                "builtInAdapter": {
                    "memBufferBytes": 134217728,
                    "modelLoadingTimeoutMillis": 90000,
                    "runtimeManagementPort": 8888,
                    "serverType": "ovms"
                },
                "containers": [
                    {
                        "args": [
                            "--port=8001",
                            "--rest_port=8888",
                            "--config_path=/models/model_config_list.json",
                            "--file_system_poll_wait_seconds=0",
                            "--grpc_bind_address=127.0.0.1",
                            "--rest_bind_address=127.0.0.1"
                        ],
                        "image": "registry.redhat.io/rhods/odh-openvino-servingruntime-rhel8@sha256:8af20e48bb480a7ba1ee1268a3cf0a507e05b256c5fcf988f8e4a3de8b87edc6",
                        "name": "ovms",
                        "resources": {
                            "limits": {
                                "cpu": "2",
                                "memory": "8Gi"
                            },
                            "requests": {
                                "cpu": "1",
                                "memory": "4Gi"
                            }
                        }
                    }
                ],
                "grpcDataEndpoint": "port:8001",
                "grpcEndpoint": "port:8085",
                "multiModel": true,
                "protocolVersions": [
                    "grpc-v1"
                ],
                "replicas": 1,
                "supportedModelFormats": [
                    {
                        "autoSelect": true,
                        "name": "openvino_ir",
                        "version": "opset1"
                    },
                    {
                        "autoSelect": true,
                        "name": "onnx",
                        "version": "1"
                    }
                ]
            }
        }
    ],
    "kind": "ServingRuntimeList",
    "metadata": {
        "continue": "",
        "resourceVersion": "1462210"
    }
}