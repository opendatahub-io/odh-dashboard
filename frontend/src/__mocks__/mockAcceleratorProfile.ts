import { AcceleratorProfileKind } from '~/k8sTypes';
import { Toleration, TolerationEffect, TolerationOperator } from '~/types';
import { genUID } from './mockUtils';

type MockResourceConfigType = {
  name?: string;
  namespace?: string;
  uid?: string;
  displayName?: string;
  identifier?: string;
  description?: string;
  enabled?: boolean;
  tolerations?: Toleration[];
};

export const mockAcceleratorProfile = ({
  name = 'migrated-gpu',
  namespace = 'test-project',
  uid = genUID('service'),
  displayName = 'Nvidia GPU',
  identifier = 'nvidia.com/gpu',
  description = '',
  enabled = true,
  tolerations = [
    {
      key: 'nvidia.com/gpu',
      operator: TolerationOperator.EXISTS,
      effect: TolerationEffect.NO_SCHEDULE,
    },
  ],
}: MockResourceConfigType): AcceleratorProfileKind => ({
  apiVersion: 'dashboard.opendatahub.io/v1',
  kind: 'AcceleratorProfile',
  metadata: {
    creationTimestamp: '2023-03-17T16:12:41Z',
    generation: 1,
    name,
    namespace,
    resourceVersion: '1309350',
    uid,
  },
  spec: {
    identifier,
    displayName,
    enabled,
    tolerations,
    description,
  },
});

export const mockGlobalScopedAcceleratorProfiles = [
  mockAcceleratorProfile({
    name: 'small-profile-global',
    displayName: 'Small Profile Global',
    namespace: 'opendatahub',
    tolerations: [
      {
        effect: TolerationEffect.NO_SCHEDULE,
        key: 'NotebooksOnlyChange',
        operator: TolerationOperator.EXISTS,
      },
    ],
  }),
  mockAcceleratorProfile({
    name: 'large-profile-global',
    displayName: 'Large Profile Global',
    namespace: 'opendatahub',
  }),
];

export const mockProjectScopedAcceleratorProfiles = [
  mockAcceleratorProfile({
    name: 'small-profile',
    displayName: 'Small Profile',
    namespace: 'test-project',
    tolerations: [
      {
        effect: TolerationEffect.NO_SCHEDULE,
        key: 'NotebooksOnlyChange',
        operator: TolerationOperator.EXISTS,
      },
    ],
  }),
  mockAcceleratorProfile({
    name: 'large-profile-1',
    displayName: 'Large Profile-1',
    namespace: 'test-project',
  }),
];
