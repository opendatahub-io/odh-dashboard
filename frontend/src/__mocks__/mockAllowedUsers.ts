import { AllowedUser, PrivilegeState } from '#~/pages/notebookController/screens/admin/types';

type MockAllowedUsersType = {
  username?: string;
  privilege?: PrivilegeState;
  lastActivity?: string;
};
export const mockAllowedUsers = ({
  username = 'test-user',
  privilege = PrivilegeState.USER,
  lastActivity = '2024-02-14T14:22:05Z',
}: MockAllowedUsersType): AllowedUser => ({
  username,
  privilege,
  lastActivity,
});
