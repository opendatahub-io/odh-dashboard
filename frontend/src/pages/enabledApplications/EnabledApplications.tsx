import * as React from 'react';
import * as _ from 'lodash';
import { Gallery, PageSection } from '@patternfly/react-core';
import { useWatchComponents } from '../../utilities/useWatchComponents';
import { OdhApplication } from '../../types';
import ApplicationsPage from '../ApplicationsPage';
import OdhAppCard from '../../components/OdhAppCard';
import QuickStarts from '../../app/QuickStarts';

import './EnabledApplications.scss';

const description = `Launch your enabled applications or get started with quick start instructions
 and tasks.`;

type EnabledApplicationsInnerProps = {
  loaded: boolean;
  loadError?: Error;
  components: OdhApplication[];
};
export const EnabledApplicationsInner: React.FC<EnabledApplicationsInnerProps> = React.memo(
  ({ loaded, loadError, components }) => {
    const isEmpty = !components || components.length === 0;
    return (
      <ApplicationsPage
        title="Enabled"
        description={description}
        loaded={loaded}
        empty={isEmpty}
        loadError={loadError}
      >
        {!isEmpty ? (
          <div className="odh-dashboard__page-content">
            <PageSection>
              <Gallery className="odh-installed-apps__gallery" hasGutter>
                {components.map((c) => (
                  <OdhAppCard key={c.metadata.name} odhApp={c} />
                ))}
              </Gallery>
            </PageSection>
          </div>
        ) : null}
      </ApplicationsPage>
    );
  },
);
EnabledApplicationsInner.displayName = 'EnabledApplicationsInner';

const EnabledApplications: React.FC = () => {
  const { components, loaded, loadError } = useWatchComponents(true);

  const sortedComponents = React.useMemo(() => {
    return _.cloneDeep(components).sort((a, b) =>
      a.spec.displayName.localeCompare(b.spec.displayName),
    );
  }, [components]);

  return (
    <QuickStarts>
      <EnabledApplicationsInner
        loaded={loaded}
        components={sortedComponents}
        loadError={loadError}
      />
    </QuickStarts>
  );
};

export default EnabledApplications;
