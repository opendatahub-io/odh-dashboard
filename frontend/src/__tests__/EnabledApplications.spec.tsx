import * as React from 'react';
import { mount } from 'enzyme';
import { EnabledApplicationsInner } from '../pages/enabledApplications/EnabledApplications';
import { mockEnabledApplications } from '../../__mocks__/mockEnabledApplications';

describe('EnabledApplications', () => {
  it('should display a message when there are no enabled applications', () => {
    const wrapper = mount(<EnabledApplicationsInner loaded={true} components={[]} />);
    expect(wrapper.find('[data-test-id="empty-empty-state"]').exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should display a message when loading applications fails', () => {
    const wrapper = mount(
      <EnabledApplicationsInner
        loaded={true}
        loadError={new Error('faux error')}
        components={[]}
      />,
    );
    expect(wrapper.find('[data-test-id="error-empty-state"]').exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should display enabled applications', () => {
    const wrapper = mount(
      <EnabledApplicationsInner loaded={true} components={mockEnabledApplications} />,
    );
    expect(wrapper.find('.odh-dashboard__page-content').exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
});
