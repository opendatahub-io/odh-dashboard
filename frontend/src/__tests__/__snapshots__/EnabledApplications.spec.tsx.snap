// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnabledApplications should display a message when loading applications fails 1`] = `
"<section class=\\"pf-c-page__main-section pf-m-light odh-apps__heading\\"><div class=\\"pf-c-content odh-apps__heading__text\\"><h1 data-pf-content=\\"true\\" class=\\"\\">Enabled</h1><p data-pf-content=\\"true\\" class=\\"\\">Launch your enabled applications or get started with quick start instructions
 and tasks.</p></div></section><section class=\\"pf-c-page__main-section\\"><div class=\\"pf-c-empty-state\\" data-test-id=\\"error-empty-state\\"><div class=\\"pf-c-empty-state__content\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 1088 1024\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"pf-c-empty-state__icon\\"><path d=\\"M1057.10141,663.5 L845.101405,215.4 C787.101405,71.8 665.401405,0 542.901405,0 C420.201405,0 296.701405,71.9 235.001405,215.6 L31.7014051,648.5 C10.4014051,700 -0.0985948775,752.3 0.000697596367,800.8 C0.301405123,924.8 70.2014051,1024 209.101405,1024 L868.401405,1024 C1005.80141,1024 1087.70141,918.6 1088.00215,795.5 C1088.10141,752.4 1078.20141,707.2 1057.10141,663.5 Z M959.401405,800.3 C958.701405,822.9 952.901405,843.5 942.601405,859.7 C926.801405,884.6 902.601405,896.7 868.301405,896.7 L209.101405,896.7 C191.201405,896.7 176.601405,893.8 165.401405,888.2 C157.301405,884 150.801405,878.4 145.401405,870.3 C135.101405,855 129.101405,832 128.401405,805.6 C127.601405,772.8 134.901405,736.5 149.401405,700.5 L353.001405,266.7 C363.201405,242.9 376.101405,221.5 391.101405,203.2 C404.801405,186.6 420.301405,172.4 437.401405,161.1 C469.201405,139.9 505.701405,128.8 542.901405,128.8 C579.701405,128.8 615.401405,139.8 646.001405,160.5 C662.401405,171.6 677.101405,185.4 690.101405,201.6 C704.501405,219.6 716.401405,240.6 725.901405,264 L940.901405,718.9 L941.101405,719.3 L941.301405,719.7 C953.901405,746 960.201405,773.9 959.401405,800.3 Z M586.601405,832 L501.301405,832 C489.501405,831.8 480.201405,821.5 480.001405,808.7 L480.001405,727.3 C480.201405,714.5 489.601405,704.3 501.301405,704 L586.601405,704 C598.401405,704.2 607.701405,714.5 607.901405,727.3 L607.901405,808.7 L608.001405,808.7 C607.701405,821.5 598.301405,831.8 586.601405,832 M639.901405,290.7 L613.201405,610.4 C611.801405,626.9 598.001405,640 581.301405,640 L506.601405,640 C490.001405,640 476.101405,627.2 474.701405,610.7 L448.101405,291 C446.501405,272.3 461.301405,256.3 480.001405,256.3 L608.001405,256 C626.701405,256 641.401405,272 639.901405,290.7\\"></path></svg><h5 class=\\"pf-c-title pf-m-lg\\">Error loading components</h5><div class=\\"pf-c-empty-state__body odh-dashboard__error-body\\"><div><code class=\\"odh-dashboard__display-error\\">faux error</code></div></div></div></div></section>"
`;

exports[`EnabledApplications should display a message when there are no enabled applications 1`] = `
"<section class=\\"pf-c-page__main-section pf-m-light odh-apps__heading\\"><div class=\\"pf-c-content odh-apps__heading__text\\"><h1 data-pf-content=\\"true\\" class=\\"\\">Enabled</h1><p data-pf-content=\\"true\\" class=\\"\\">Launch your enabled applications or get started with quick start instructions
 and tasks.</p></div></section><section class=\\"pf-c-page__main-section pf-m-fill\\"><div class=\\"pf-c-empty-state\\" data-test-id=\\"empty-empty-state\\"><div class=\\"pf-c-empty-state__content\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\" class=\\"pf-c-empty-state__icon\\"><path d=\\"M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z\\"></path></svg><h5 class=\\"pf-c-title pf-m-lg\\">No Components Found</h5></div></div></section>"
`;

exports[`EnabledApplications should display enabled applications 1`] = `
"<section class=\\"pf-c-page__main-section pf-m-light odh-apps__heading\\"><div class=\\"pf-c-content odh-apps__heading__text\\"><h1 data-pf-content=\\"true\\" class=\\"\\">Enabled</h1><p data-pf-content=\\"true\\" class=\\"\\">Launch your enabled applications or get started with quick start instructions
 and tasks.</p></div></section><div class=\\"odh-dashboard__page-content\\"><section class=\\"pf-c-page__main-section\\"><div class=\\"pf-l-gallery pf-m-gutter odh-installed-apps__gallery\\"><article id=\\"jupyterhub\\" class=\\"pf-c-card pf-m-hoverable pf-m-selectable odh-card odh-tourable-card\\" tabindex=\\"0\\" data-ouia-component-type=\\"PF4/Card\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Card-1\\"><div class=\\"pf-c-card__header\\"><img class=\\"pf-c-brand odh-card__header-brand\\" src=\\"images/jupyterhub.svg\\" alt=\\"JupyterHub\\"><div class=\\"pf-c-dropdown pf-m-align-right\\" data-ouia-component-type=\\"PF4/Dropdown\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Dropdown-1\\"><button aria-label=\\"Actions\\" id=\\"pf-dropdown-toggle-id-0\\" class=\\"pf-c-dropdown__toggle pf-m-plain\\" type=\\"button\\" aria-expanded=\\"false\\" aria-haspopup=\\"true\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 192 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z\\"></path></svg></button></div></div><div class=\\"pf-c-card__title\\"><span class=\\"odh-card__title-supported\\">JupyterHub</span></div><div class=\\"pf-c-card__body\\">A multi-user version of the notebook designed for companies, classrooms and research labs.</div><div class=\\"pf-c-card__footer odh-card__footer\\"><a class=\\"odh-card__footer__link\\" href=\\"https://jupyterhub-redhat-ods-applications.apps.jephilli-4-9-06-21-0953.devcluster.openshift.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Launch<svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z\\"></path></svg></a><a class=\\"odh-card__footer__link\\" href=\\"#\\">Start</a></div></article></div></section></div>"
`;
